src/Listener/HttpPoolWatcher.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Hyperf + OpenCodeCo
+ *
+ * @link     https://opencodeco.dev
+ * @document https://hyperf.wiki
+ * @contact  leo@opencodeco.dev
+ * @license  https://github.com/opencodeco/hyperf-metric/blob/main/LICENSE
+ */
+
+namespace Hyperf\Metric\Listener;
+
+use Hyperf\Contract\ConfigInterface;
+use Hyperf\Contract\ContainerInterface;
+use Hyperf\Coordinator\Timer;
+use Hyperf\Event\Contract\ListenerInterface;
+use Hyperf\Framework\Event\BeforeWorkerStart;
+use Hyperf\Metric\Metric;
+use Hyperf\Pool\SimplePool\PoolFactory;
+use Hyperf\Server\Event\MainCoroutineServerStart;
+
+class HttpPoolWatcher implements ListenerInterface
+{
+    protected Timer $timer;
+
+    public function __construct(protected ContainerInterface $container)
+    {
+        $this->timer = new Timer();
+    }
+
+    public function listen(): array
+    {
+        return [
+            BeforeWorkerStart::class,
+            MainCoroutineServerStart::class,
+        ];
+    }
+
+    public function process(object $event): void
+    {
+        /** @var PoolFactory $factory */
+        $factory = $this->container->get(PoolFactory::class);
+        $worker = (string) ($event->workerId ?? 0);
+
+        $config = $this->container->get(ConfigInterface::class);
+
+        $timer_interval = $config->get('metric.default_metric_interval', 5);
+
+        $this->timer->tick($timer_interval, function () use ($factory, $worker) {
+            foreach ($factory->getPoolNames() as $name) {
+                $pool = $factory->get($name, function () {});
```

Comment:
The `PoolFactory::get` method returns the existing pool or creates one based on the provided parameters (callback and options). The listener retrieves existing pools before requesting them, thereby reducing the likelihood of creating a new connection pool. However, to completely neutralize this risk, there exists another implementation possibility using Reflection.
```php
public function process(object $event): void
{
    /** @var PoolFactory $factory */
    $factory = $this->container->get(PoolFactory::class);
    $worker = (string) ($event->workerId ?? 0);

    $config = $this->container->get(ConfigInterface::class);

    $timer_interval = $config->get('metric.default_metric_interval', 5);

    $this->timer->tick($timer_interval, function () use ($factory, $worker) {
        $pools = $this->getPools($factory);
        foreach ($factory->getPoolNames() as $name) {
            $labels = [
                'worker' => $worker,
                'pool' => implode('.', array_slice(explode('.', (string) $name), 2, -2)),
            ];

            Metric::gauge('http_connections_in_use', (float) $pools[$name]->getCurrentConnections(), $labels);
            Metric::gauge('http_connections_in_waiting', (float) $pools[$name]->getConnectionsInChannel(), $labels);
            Metric::gauge('http_max_connections', (float) $pools[$name]->getOption()->getMaxConnections(), $labels);
        }
    });
}

/**
 * @return array<string, Pool>
 */
private function getPools(PoolFactory $factory): array
{
    $reflection = new ReflectionClass($factory::class);
    $property = $reflection->getProperty('pools');
    $property->setAccessible(true);

    return $property->getValue($factory);
}
```